src/components/MarkdownActionsDropdown/index.tsx

```diff
@@ -0,0 +1,120 @@
+import { Menu } from '@headlessui/react'
+import { IconCopy, IconEye, IconExternal } from '@posthog/icons'
+import { Chevron } from 'components/Icons'
+import CheckIcon from '../../images/check.svg'
+import React, { useState } from 'react'
+
+interface CopyMarkdownActionsDropdownProps {
+    /** The markdown content to work with */
+    markdownContent: string
+    /** Page URL for context */
+    pageUrl: string
+}
+
+export const CopyMarkdownActionsDropdown: React.FC<CopyMarkdownActionsDropdownProps> = ({
+    markdownContent,
+    pageUrl,
+}) => {
+    const [copiedState, setCopiedState] = useState<string | null>(null)
+
+    // Consolidated styles
+    const menuItemButtonStyles =
+        '!m-0 py-1.5 px-3 !text-sm cursor-pointer rounded-sm hover:bg-light active:bg-accent dark:hover:bg-light/10 dark:active:bg-light/5 transition-colors hover:transition-none whitespace-nowrap w-full text-left flex items-center'
+    const menuItemIconStyles = 'w-4 h-4 mr-3'
+
+    // Helper function to strip query parameters from URL
+    const getCleanPath = (url: string) => {
+        return url.split('?')[0]
+    }
+
+    const handleCopyMarkdown = () => {
+        navigator.clipboard.writeText(markdownContent)
+        setCopiedState('markdown')
+        setTimeout(() => setCopiedState(null), 2000)
+    }
+
+    const handleViewAsMarkdown = () => {
+        const cleanPath = getCleanPath(pageUrl)
+        const markdownUrl = `${cleanPath}.md`
+        window.open(markdownUrl, '_blank')
```

Comment:
Is this really the nest navigation logic here? I actually don't know, but window.open is always suspect to me 😆